### PR TITLE
node: podresources: graduate to GA

### PR DESCRIPTION
**Pull request by ffromani**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Complete the graduation of the podresources endpoint, by locking the FG to ON

#### Which issue(s) this PR fixes:
Fixes https://github.com/kubernetes/enhancements/issues/3743

#### Special notes for your reviewer:
Depends on:
- https://github.com/kubernetes/kubernetes/pull/115133
- https://github.com/kubernetes/kubernetes/pull/116459

#### Does this PR introduce a user-facing change?
```release-note
The kubelet podresources endpoint is GA and always enabled
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
Please use the following format for linking documentation:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/issues/606
```

### CODE REVIEW

**Comment by ffromani** [comment]
/hold

because deps haven't merged and will likely need a rebase after them merge

File: cmd/kubelet/app/server.go
Comment (bart0sh) [inline cmd/kubelet/app/server.go:1196]: Should this be done in 1.29?

Comment (ffromani) [reply cmd/kubelet/app/server.go:1196]: AFAIK the change with locks the FG to on is also supposed to remove the corresponding if()s - we did so when promoting cpumanager/devicemanager and if memory serves in other GA graduations, but if I got it wrong I'm happy to revert the FG check removals.

Comment (SergeyKanzhelev) [reply cmd/kubelet/app/server.go:1196]: when locked to default we remove all checks. Lock to default is only for not breaking users passing feature gates into the command line or kubelet config

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

**Comment by ffromani** [comment]
rebased to fix conflicts

**Comment by ffromani** [comment]
/retest

failed some storage unit test, unrelated to this PR

**Comment by ffromani** [comment]
/hold

thanks @bart0sh for LGTM! I'm holding because I'm not sure process-wise I need to update the KEP before (being this a GA graduation). Later today we have the sig-nod weekly, I'll clarify asking during the meeting and/or asking in the channel, and unhold ASAP.

**Comment by ffromani** [comment]
> /hold
> 
> thanks @bart0sh for LGTM! I'm holding because I'm not sure process-wise I need to update the KEP before (being this a GA graduation). Later today we have the sig-node weekly, I'll clarify by asking during the meeting and/or asking in the channel, and unhold ASAP.

xref: https://github.com/kubernetes/enhancements/pull/3980

**Review by swatisehgal — APPROVED** [review]
/lgtm

**Review by pacoxu — APPROVED** [review]

**Comment by SergeyKanzhelev** [comment]
/unhold

(based on https://github.com/kubernetes/enhancements/pull/3980#issuecomment-1563553562). Some more paperwork may be needed (like marking the KEP `implemented`), but we had enough discussions to have this portion to GA already. Thank you @ffromani !

**Pull request merged**
